common/test/common_tests/logic/components_touched_test.cljc

```diff
@@ -0,0 +1,151 @@
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns common-tests.logic.components-touched-test
+  (:require
+   [app.common.files.changes-builder :as pcb]
+   [app.common.files.libraries-helpers :as cflh]
+   [clojure.test :as t]
+   [common-tests.helpers.compositions :as tho]
+   [common-tests.helpers.files :as thf]
+   [common-tests.helpers.ids-map :as thi]))
+
+(t/use-fixtures :each thi/test-fixture)
+
+(t/deftest test-touched-when-changing-attribute
+  (let [;; Setup
+        file  (-> (thf/sample-file :file1)
+                  (tho/add-simple-component-with-copy :component1
+                                                      :main-root
+                                                      :main-child
+                                                      :copy-root
+                                                      :main-child-params {:fills (thf/sample-fills-color
+                                                                                  :fill-color "#abcdef")}))
+        page      (thf/current-page file)
+        copy-root (thf/get-shape file :copy-root)
+
+        ;; Action
+        update-fn (fn [shape]
+                    (assoc shape :fills (thf/sample-fills-color :fill-color "#fabada")))
+
+        changes   (cflh/generate-update-shapes (pcb/empty-changes nil (:id page))
+                                               (:shapes copy-root)
+                                               update-fn
+                                               (:objects page)
+                                               {})
+
+        file'     (thf/apply-changes file changes)
+
+        ;; Get
+        copy-root'  (thf/get-shape file' :copy-root)
+        copy-child' (thf/get-shape-by-id file' (first (:shapes copy-root')))
+        fills'      (:fills copy-child')
+        fill'       (first fills')]
+
+    ;; Check
+    (t/is (= (count fills') 1))
+    (t/is (= (:fill-color fill') "#fabada"))
+    (t/is (= (:fill-opacity fill') 1))
+    (t/is (= (:touched copy-root') nil))
+    (t/is (= (:touched copy-child') #{:fill-group}))))
+
+(t/deftest test-not-touched-when-adding-shape
+  (let [;; Setup
+        file (-> (thf/sample-file :file1)
+                 (tho/add-simple-component-with-copy :component1
+                                                     :main-root
+                                                     :main-child
+                                                     :copy-root)
+                 (thf/add-sample-shape :free-shape))
+
+        page (thf/current-page file)
+
+        ;; Action
+        ;; IMPORTANT: as modifying copies structure is now forbidden, this action
+        ;; will not have any effect, and so the parent shape won't also be touched.
+        changes (cflh/generate-relocate-shapes (pcb/empty-changes)
+                                               (:objects page)
+                                               #{(thi/id :copy-root)}   ; parents
```

Comment:
This should be the parent of the shape that you are moving: `#{(:parent-id (thf/get-shape :free-shape))}` Or in this case `uuid/zero`